### PR TITLE
Convert RTCD to deployments from Daemonsets

### PR DESCRIPTION
**Pull request by d-wierdsma**
#### Summary

Convert RTCD to deployments from Daemonsets

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-5537

### CODE REVIEW

File: charts/mattermost-rtcd/values.yaml [hunk 1]
```diff
@@ -95,4 +97,15 @@ tolerations: {}
 
 dnsConfig: {}
 
-affinity: {}
+affinity:
```
Comment (stylianosrigas) [inline charts/mattermost-rtcd/values.yaml:100]: @d-wierdsma We need an affinity here that ensures that not both pods will be scheduled in the same rtcd node as calls will not work with that.

Comment (d-wierdsma) [reply charts/mattermost-rtcd/values.yaml:100]: Updated change

**Review by stafot — APPROVED** [review]

**Comment by stafot** [comment]
You could make replicas number configurable but 0/5.

**Comment by d-wierdsma** [comment]
> You could make replicas number configurable but 0/5.

@stafot I believe I did here https://github.com/mattermost/mattermost-helm/pull/397/files#diff-0e226ccb0251623c490fd79a8dfa867c8039a9a99dd6e1fba48b053a3f3020c0R37

**Review by angeloskyratzakos — APPROVED** [review]

**Comment by streamer45** [comment]
@d-wierdsma Do any of these changes require updating our documentation in https://docs.mattermost.com/configure/calls-deployment.html#kubernetes-deployments?

**Comment by d-wierdsma** [comment]
Ah shoot yes, can you link me where those docs are stored and I can update them?

**Comment by d-wierdsma** [comment]
Thanks for bringing that to my attention Claudio!

Here is the PR https://github.com/mattermost/docs/pull/6350

**Comment by stafot** [comment]
@d-wierdsma Since we had as default method of deploying calls `daemonSet` in my view we should support it for sometime as might other people use it. Since this PR introduces a breaking change a safer approach would be to support both types of  deploying resources (`daemonsets` and `deployments`). Initially we could leave as default `daemonset` and be able with a flag to use `deployment` and at later step do default deployment and second way `daemonset`. I apologize because I did this comment after the merge of this code.

**Pull request merged**